File: backend/src/main/java/nlu/fit/backend/dto/user/request/UpdateUserRequest.java
package nlu.fit.backend.dto.user.request;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class UpdateUserRequest {
    @NotBlank(message = "Họ và tên không được để trống")
    private String fullName;
    @NotBlank(message = "Giới tính không được để trống")
    private String gender;
    @NotBlank(message = "Số điện thoại không được để trống")
    private String phone;
    private String avatarUrl;
    private String avatarPublicId;
}

File: frontend/lib/models/user/update_user_request.dart
class UpdateUserRequest {
  final String fullName;
  final String phone;
  final String gender;
  final String? avatarUrl;
  final String? avatarPublicId;

  UpdateUserRequest({
    required this.fullName,
    required this.phone,
    required this.gender,
    this.avatarUrl,
    this.avatarPublicId
  });

  Map<String, dynamic> toJson() {
    return {
      'fullName': fullName,
      'phone': phone,
      'gender': gender,
      if (avatarUrl != null) 'avatarUrl': avatarUrl,
      if (avatarPublicId != null) 'avatarPublicId': avatarPublicId,
    };
  }
}